compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -1213,8 +1220,25 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                 deny_equality_constraints(self, predicate, generics);
             }
         }
+        walk_list!(self, visit_generic_param, &generics.params);
+        for predicate in &generics.where_clause.predicates {
+            match predicate {
+                WherePredicate::BoundPredicate(bound_pred) => {
+                    // A type binding, eg `for<'c> Foo: Send+Clone+'c`
+                    self.check_late_bound_lifetime_defs(&bound_pred.bound_generic_params);
 
-        visit::walk_generics(self, generics)
+                    self.visit_ty(&bound_pred.bounded_ty);
+
+                    self.trait_ref_hack = !bound_pred.bound_generic_params.is_empty();
```

Comment:
I initially imagined that instead of using a flag, we would match on `bound_pred` to see whether it contained a nested quantifier. That would involve duplicating a bit of code, though.

Reply:
That was easy enough to do :)